src/test/FuzzerImpl.cpp

```diff
@@ -89,9 +115,11 @@ getShortKey(Asset const& asset)
     case ASSET_TYPE_NATIVE:
         return 0;
     case ASSET_TYPE_CREDIT_ALPHANUM4:
-        return getShortKey(asset.alphaNum4().issuer);
+        return encodeAsset(getShortKey(asset.alphaNum4().issuer),
```

Comment:
> So what is the significance of changing how you pull out the bytes?

It's not necessary for correctness that it be that particular thing, but given that flexibility, I think the least surprising code would be what makes "compact" a left-inverse of "unpack".  Is there anything else that would be better?  (I'm considering "reads unsurprisingly now" to be more important than "doesn't change what was there before".)

Reply:
I'm not following -- it seems to me that we do want the _decoding_ (`decodeAsset()`, `makeAsset()`)  to be able to generate `Asset`s with `issuer != digit`, which is most of the change, so that we can fuzz-test multiple `Asset`s with the same issuer.  Reverting the decoding change would change the semantics back to the old ones, re-introducing that inflexibility.  Reverting the _encoding_ part would not change the semantics, as you correctly point out, but it would only delete the 6-line `encodeAsset()` function (or replace the mask/shift/or with the old `[0]` in `encodeAsset()`).  Or am I missing an opportunity to delete more stuff?

Reply:
I didn't look closely enough at the line number when I wrote that; I had forgotten that some of the change was related to the decoding functionality which you indeed do need. You are correct that I overstated the complexity. A better estimate is about 8-10 lines: 6 for `encodeAsset`, 2 for reverting the calls to `encodeAsset`, and I'm not sure you need all the constants anymore.

Reply:
I could get rid of `ENCODE_ASSET_ISSUER_MASK` and `NUMBER_OF_ISSUERS_TO_USE`, so it'd be 11 fewer lines (including one whitespace), minus probably a couple for a comment.  Is that still your preference?

Reply:
I mean for me the main benefit is removing `encodeAsset`, so I still support it. Future maintainers will look at it and either (a) perpetuate that complexity for no benefit or (b) be perplexed about why it exists at all.

Reply:
Okay, that works for me.  I've pushed another commit to remove `encodeAsset()`.